src/slider/index.ts

```diff
@@ -0,0 +1,140 @@
+import { tag, utils, Tag } from '@storefront/core';
+
+@tag('gb-slider', require('./index.html'), require('./index.css'))
+class Slider {
+
+  props: Slider.Props = {
+    defaultLower: 0,
+    defaultUpper: 100
+  };
+
+  onMount() {
+    this.state = {
+      ...this.state,
+      handleLower: this.root.querySelector('.gb-slider__handle-lower'),
+      handleUpper: this.root.querySelector('.gb-slider__handle-upper'),
+      connect: this.root.querySelector('.gb-slider__connect')
+    };
+  }
+
+  onClick = (event: MouseEvent & Slider.Event) => {
+    event.preventDefault();
+    event.stopPropagation();
+    const handleUpper = this.state.handleUpper;
+    const handleLower = this.state.handleLower;
+    const handleLowerLeft = handleLower.getBoundingClientRect().left;
+    const handleUpperRight = handleUpper.getBoundingClientRect().right;
+    const mouseLeft = event.clientX;
+    if (Math.abs(mouseLeft - handleLowerLeft) < Math.abs(mouseLeft - handleUpperRight)) {
+      this.setup(handleLower);
+    } else {
+      this.setup(handleUpper);
+    }
+    this.state = { ...this.state, mouseLeft };
+    this.moveElements();
```

Comment:
if we always set `this.state = { ...this.state, mouseLeft: <something> };` before calling `this.moveElements();`, should we maybe make `moveElements()` set the state?